components/fxa-client/src/device.rs

```diff
@@ -162,19 +169,30 @@ impl FirefoxAccount {
     /// Poll and parse any pending available command for our device.
     /// This should be called semi-regularly as the main method of
     /// commands delivery (push) can sometimes be unreliable on mobile devices.
+    /// Typically called even when a push notification is received, so that
+    /// any prior messages for which a push didn't arrive are still handled.
     ///
     /// **💾 This method alters the persisted account state.**
-    pub fn poll_device_commands(&mut self) -> Result<Vec<IncomingDeviceCommand>> {
+    pub fn poll_device_commands(
+        &mut self,
+        reason: &CommandFetchReason,
+    ) -> Result<Vec<IncomingDeviceCommand>> {
         let last_command_index = self.state.last_handled_command.unwrap_or(0);
         // We increment last_command_index by 1 because the server response includes the current index.
-        self.fetch_and_parse_commands(last_command_index + 1, None)
+        self.fetch_and_parse_commands(last_command_index + 1, None, reason)
     }
 
     /// Retrieve and parse a specific command designated by its index.
     ///
     /// **💾 This method alters the persisted account state.**
-    pub fn fetch_device_command(&mut self, index: u64) -> Result<IncomingDeviceCommand> {
-        let mut device_commands = self.fetch_and_parse_commands(index, Some(1))?;
+    ///
+    /// Note that this should not be used if possible, as it does not correctly
+    /// handle missed messages. It's currently used only on iOS due to platform
```

Comment:
Out of curiosity, what does it do that's bad? 😅 Is that the "1 command per push message" limitation that the `push.rs` docs mention?

Reply:
It will not see missed messages. If I get a push for `tab-1` and a push for `tab-3`, and between them I've never explicitly polled, I'll miss `tab-2`, even if I try polling now.

Reply:
I added that to the comment :)